ext/string_metric/levenshtein/iterative_with_two_matrix_rows_ext.c

```diff
@@ -0,0 +1,99 @@
+#include <ruby.h>
+
+#define MIN(a, b, c) ((a) < (b) ? ((a) < (c) ? (a) : (c)) : ((b) < (c) ? (b) : (c)))
+
+#define MALLOC_W(ptr, size) \
+    if ((ptr = malloc(size)) == NULL) \
+        rb_memerror()
+
+VALUE distance_ext(VALUE self, VALUE v_from, VALUE v_to, VALUE v_from_len, VALUE v_to_len,
+                   VALUE v_max_distance, VALUE v_insertion_cost, VALUE v_deletion_cost,
+                   VALUE v_substitution_cost);
+
+VALUE StringMetric = Qnil;
+VALUE Levenshtein = Qnil;
+VALUE IterativeWithTwoMatrixRowsExt = Qnil;
+
+void
+Init_iterative_with_two_matrix_rows_ext(void)
+{
+    StringMetric = rb_define_module("StringMetric");
+    Levenshtein = rb_define_module_under(StringMetric, "Levenshtein");
+
+    IterativeWithTwoMatrixRowsExt =
+        rb_define_class_under(Levenshtein, "IterativeWithTwoMatrixRowsExt", rb_cObject);
+
+    rb_define_singleton_method(IterativeWithTwoMatrixRowsExt, "distance_ext", distance_ext, 8);
+}
+
+VALUE
+distance_ext(VALUE self, VALUE v_from, VALUE v_to, VALUE v_from_len, VALUE v_to_len,
+             VALUE v_max_distance, VALUE v_insertion_cost, VALUE v_deletion_cost,
+             VALUE v_substitution_cost)
+{
+    /* Ruby to C datatype conversions & variables declaration-initialization */
+    long max_distance =      NUM2LONG(v_max_distance);
+    long insertion_cost =    NUM2LONG(v_insertion_cost);
+    long deletion_cost =     NUM2LONG(v_deletion_cost);
+    long substitution_cost = NUM2LONG(v_substitution_cost);
```

Comment:
Small optimization opportunity here by using FIX2INT instead of NUM2LONG in the program.
They are faster and will provide an exception if the conversion would overflow the integer.
If you always want to be correct and be able to handle extreme values, you are much better of leaving everything as is. :)